tests/unit/_internal/models/test_model.py

```diff
@@ -0,0 +1,164 @@
+import os
+from typing import TYPE_CHECKING
+from datetime import datetime
+from datetime import timezone
+
+import fs
+import pytest
+import fs.errors
+
+from bentoml import Tag
+from bentoml.exceptions import BentoMLException
+from bentoml._internal.models.model import Model
+from bentoml._internal.models.model import ModelInfo
+from bentoml._internal.configuration import BENTOML_VERSION
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+expected_yaml = (
+    "name: test\n"
+    "version: version\n"
+    "bentoml_version: {bentoml_version}\n"
+    "creation_time: {creation_time}\n"
+    "api_version: v1\n"
+    "module: testmodule\n"
+    "context:\n"
+    "  ctx: 1\n"
+    "labels:\n"
+    "  label: stringvalue\n"
+    "options:\n"
+    "  option:\n"
+    "    dictkey: dictvalue\n"
+    "metadata:\n"
+    "  metadatakey:\n"
+    "  - list\n"
+    "  - of\n"
+    "  - str\n"
+)
```

Comment:
```suggestion
expected_yaml = """
name: test
version: version
bentoml_version: {bentoml_version}
creation_time: {creation_time}
api_version: v1
module: testmodule
context:
  ctx: 1
labels:
  label: stringvalue
options:
  option:
    dictkey: dictvalue
metadata:
  metadatakey:
  - list
  - of
  - str
"""
```

something like this?